patches/master_patch.patch

```diff
@@ -1844,6 +1844,36 @@ index a433afd0217830b49ddb3381ad55afeae2381840..ddcca749666f956c6317676dbcdd48da
    DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
  }
  
+diff --git a/net/url_request/url_request_http_job.cc b/net/url_request/url_request_http_job.cc
+index da7a19865989404f3147ca3e29a13f7f29f175d4..47fa22638f7450367c9014d2f2fa70f1659d2f15 100644
+--- a/net/url_request/url_request_http_job.cc
++++ b/net/url_request/url_request_http_job.cc
+@@ -415,8 +415,10 @@ void URLRequestHttpJob::NotifyBeforeSendHeadersCallback(
+ }
+ 
+ void URLRequestHttpJob::NotifyHeadersComplete() {
+-  DCHECK(!response_info_);
++  std::string first_party_host = request_->site_for_cookies().host();
```

Comment:
do we care about this warning? https://cs.chromium.org/chromium/src/net/url_request/url_request.h?q=site_for_cookies&sq=package:chromium&l=270

`site_for_cookies` is used for firstPartyUrl in webRequest also; i don't exactly remember why we decided it wasn't an issue there

Reply:
Even in this case we are using `site_for_cookies` to figure out the first party url.

Reply:
according to https://tools.ietf.org/html/draft-ietf-httpbis-cookie-same-site-00 section 2.1, site for cookies is not the same as top-level site in a few cases:

1. in a nested cross-origin iframe, the site-for-cookies is ''
2. for shared workers belonging to documents with different origins, the site-for-cookies is ''

Reply:
i guess it doesn't matter in this case since we only use the first party host to check if it is the same as the request host (which would not be empty in the cases above): https://github.com/brave/muon/pull/527/files#diff-85ff0d997b1a2f089546f7b2823b0ad7R1869